### PR TITLE
new: search api

### PR DESCRIPTION
**Pull request by ItamarShDev**
### Description

search api to use Fotmob search power

### Checklist

- [x] Code compiles correctly
- [x] Created tests which fail without the change (if applicable)
- [x] All lint and unit tests passing
- [x] Extended the README / documentation, if necessary

### Additional Comments

Anything else you'd like to add.

### CODE REVIEW

File: test/e2e.test.ts
Comment (bgrnwd) [inline test/e2e.test.ts:1]: I'd prefer not to actually hit the API during tests. I know that's the goal of an e2e test but how would you feel about mocking out the calls?

Comment (olamarvel) [reply test/e2e.test.ts:1]: hey @bgrnwd seem @ItamarShDev has not seen this message. I just happpen to be free and this seem like a nice feature for the package to have. Would you merge it if i mock out the calls?

Comment (ItamarShDev) [reply test/e2e.test.ts:1]: Indeed I missed it!

Comment (ItamarShDev) [reply test/e2e.test.ts:1]: Will remove the tests tonight

Comment (olamarvel) [reply test/e2e.test.ts:1]: Alright then would love to add the functionality to my current app

Thank you

Comment (ItamarShDev) [reply test/e2e.test.ts:1]: Test removed

**Comment by ItamarShDev** [comment]
mistakenly closed it, here is the new one #494

**Pull request closed without merging**